HIP/hip-xxx.md

```diff
@@ -0,0 +1,167 @@
+
+## User stories
+
+- As a mirror node operator, I want to provide my API clients with the ability to list NFT allowances for an account that has granted allowance for a specific NFT serial number, as well as for those who have used  `approveForAll`.
+- As a mirror node operator, I want to provide my API clients with the ability to list NFT allowances for a spender who has received an allowance for a single NFT serial number, as well as via `approveForAll`.
+
+## Specification
+
+This is a HIP for adding the ability to query the allowances on NFTs from the mirror node.
+
+### Constraints
+
+When designing the NFT allowances APIs, it is important to consider certain constraints on the number of allowances and tokens. These constraints may limit the ability to query unbounded information.
+
+- An approval transaction can include a maximum of 20 allowance change operations.
+- An account can have a maximum of 100 hbar, token, and approved for all NFT allowances.
+- An account can have an unbounded number of individual NFT allowances.
+- A non-fungible token can have an unbounded number of approved for all NFT allowances.
+- A specific NFT can only have one explicit approval. The last one to be approved will take precedence.
+- When an NFT is transferred by either the owner or a spender, it results in the individual spender's allowance being cleared.
+- A specific NFT can have an unbounded number of implicit approved for all NFT allowances.
+
+### Approved for all NFT Allowances API
+
+This API takes a path parameter that represents either the owner or spender, depending on a boolean flag provided as a query parameter called `owner`. If the `owner` value is true, the `accountId` path parameter should indicate the ID of the owner. Conversely, if the `owner` value is false, the `accountId` path parameter should indicate the ID of the spender who has an allowance.
+
+Following is an example response of the NFT allowance API:
+
+GET `/api/v1/accounts/{accountId}/allowances/nfts`
+
+```json
+{
+  "allowances": [
+    {
+      "approved_for_all": true,
+      "owner": "0.0.1000",
+      "spender": "0.0.8488",
+      "token_id": "0.0.1033",
+      "timestamp": {
+        "from": "1633466229.96874612",
+        "to": null
+      }
+    },
+    {
+      "approved_for_all": false,
+      "owner": "0.0.1000",
+      "spender": "0.0.8488",
+      "token_id": "0.0.1034",
+      "timestamp": {
+        "from": "1633466229.96874612",
+        "to": null
+      }
+    },
+    {
+      "approved_for_all": true,
+      "owner": "0.0.1000",
+      "spender": "0.0.9857",
+      "token_id": "0.0.1032",
+      "timestamp": {
+        "from": "1633466229.96874612",
+        "to": null
+      }
+    }
+  ],
+  "links": {
+		"next": "/api/v1/accounts/0.0.1000/allowances/nfts?limit=2&order=desc&token.id=gte:0.0.1034&account.id=gt:9857"
```

Comment:
This next link doesn't match the data.
```suggestion
		"next": "/api/v1/accounts/0.0.1000/allowances/nfts?limit=3&order=asc&account.id=gte:9857&token.id=gt:0.0.1032"
```

Reply:
The limit change is correct, but I believe the account.id and token.id are correct, the order in the response isn't

Reply:
Disagree. Now the order of the items in the response is incorrect and the next link is wrong. I believe the miscommunication is that you are thinking token.id is the first column to sort, then account.id. But it should be account.id then token.id since account.id is the primary part of the index and is required when specifying a token.id.